### PR TITLE
F/orca 552 Reload notification and progress indicators, state reload/refresh fixes, container/service logs fix

### PR DESCRIPTION
**Pull request by ac111**

### CODE REVIEW

File: app/components/containerLogs/containerLogsController.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ function ($scope, $transition$, $anchorScroll, ContainerLogs, Container) {
       tail: $scope.tailLines
     }, function (data, status, headers, config) {
       // Replace carriage returns with newlines to clean up output
-      data = data.replace(/[\r]/g, '\n');
+      data = data.data.replace(/[\r]/g, '\n');
```
Comment (jakecode) [inline app/components/containerLogs/containerLogsController.js]: Is is possible to change the data parameter to a more descriptive name? data = data.data is a little nondescript when reading the code.

File: app/components/infra/infraController.js [hunk 2]
```diff
@@ -110,14 +116,27 @@ function ($interval, $q, $scope, $state, $interval, EndpointProvider, OrcaEndpoi
...
                             foundNewEndpoints.push(ep);
                             $scope.newEndpoints = foundNewEndpoints;
                             Notifications.success('Endpoint created', ep.Name);
+                            foundNewEP = true;
                         });
+                        newEPPromises.push(epPromise);
                     }
                 }
+                $q.all(newEPPromises)
+                .then(function success(data) {
```
Comment (jakecode) [inline app/components/infra/infraController.js]: Is the data parameter needed here? I don't see it used in the function attached, does javascript have a idiom that's used for parameters that are required but not used in the functions body?

File: app/components/serviceLogs/serviceLogsController.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ function ($scope, $transition$, $anchorScroll, ServiceLogs, Service) {
       tail: $scope.tailLines
     }, function (data, status, headers, config) {
       // Replace carriage returns with newlines to clean up output
-      data = data.replace(/[\r]/g, '\n');
+      data = data.data.replace(/[\r]/g, '\n');
```
Comment (jakecode) [inline app/components/serviceLogs/serviceLogsController.js]: Same as my data = data.data comment from before, is it possible to make the data parameter more descriptive?

File: app/components/stack/stackController.js [hunk 4]
```diff
@@ -73,6 +73,11 @@ function ($q, $scope, $state, $stateParams, $document, StackCreateService, Stack
     .finally(function final() {
       if (useCache == 'true') {
         $scope.stackFileContent = StackCreateService.getStackFileContent();
+
+
+        console.log("USING CACHE! Data: " + JSON.stringify($scope.stackFileContent));
```
Comment (jakecode) [inline app/components/stack/stackController.js]: This looks like it was for debugging, is it still needed?

File: app/rest/docker/containerLogs.js [hunk 5]
```diff
@@ -12,9 +12,12 @@ angular.module('portainer.rest')
           'timestamps': params.timestamps || 0,
           'tail': params.tail || 'all'
         }
-      }).success(callback).error(function (data, status, headers, config) {
-        console.log(data);
-      });
+      })
+      .then(callback);
+
+      //.catch(function error(data, status, headers, config) {
```
Comment (jakecode) [inline app/rest/docker/containerLogs.js]: Can the commented out code be removed?

File: app/rest/docker/serviceLogs.js [hunk 6]
```diff
@@ -12,9 +12,12 @@ angular.module('portainer.rest')
           'timestamps': params.timestamps || 0,
           'tail': params.tail || 'all'
         }
-      }).success(callback).error(function (data, status, headers, config) {
-        console.log(data);
-      });
+      })
+      .then(callback);
+
+      //.catch(function error(data, status, headers, config) {
```
Comment (jakecode) [inline app/rest/docker/serviceLogs.js]: Can the commented out code be removed?

File: app/components/containerLogs/containerLogsController.js (hunk 1 above)
Comment (ac111) [reply app/components/containerLogs/containerLogsController.js]: yes, agreed.  Will update

File: app/components/infra/infraController.js (hunk 2 above)
Comment (ac111) [reply app/components/infra/infraController.js]: not required, but generally accepted (from what i've seen) to accept data parameter for success callback, even if not used in the function body

File: app/components/stack/stackController.js (hunk 4 above)
Comment (ac111) [reply app/components/stack/stackController.js]: Ah, thanks for pointing this out.  Yes, purely for debugging

File: app/rest/docker/containerLogs.js (hunk 5 above)
Comment (ac111) [reply app/rest/docker/containerLogs.js]: Yes, will remove

**Pull request merged**